app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -291,13 +291,18 @@ def create_vm(args = {})
       vm
     end
 
-    def preallocate_disks(args)
-      change_allocation_volumes = args[:volumes_attributes].values.select { |x| x[:id].present? && x[:preallocate] == '1' }
-      if args[:template].present? && change_allocation_volumes.present?
-        disks = change_allocation_volumes.map do |volume|
-          { :id => volume[:id], :sparse => 'false', :format => 'raw', :storagedomain => volume[:storage_domain] }
+    def preallocate_and_clone_disks(args, template)
+      change_volumes = args[:volumes_attributes].values.select { |x| x[:id].present? }
+      if change_volumes.present?
+        disks = change_volumes.map do |volume|
+          if volume[:preallocate] == '1'
+            { :id => volume[:id], :sparse => 'false', :format => 'raw', :storage_domain => volume[:storage_domain] }
+          else
+            template_volume = template.volumes.detect { |v| v.id == volume["id"] }
+            { :id => volume["id"], :storage_domain => volume["storage_domain"] } if template_volume.storage_domain != volume["storage_domain"]
```

Comment:
If storage domain remained the same, the disk will not be changed at all. Is it intended?

Reply:
Yes, that is exactly the purpose of this.
If the storage domain changes we need to clone the disk from the template to set the new one, if it doesn't (and we don't want it preallocated) we won't send the disk or clone attribute and the disk will be created from the template as is.